cdisc_rules_engine/services/data_services/local_data_service.py

```diff
@@ -168,29 +177,37 @@ def _async_get_datasets(
             )
 
     def read_metadata(self, file_path: str) -> dict:
+        # print("local_data_service.py: reading metadata from file = ", file_path)
         file_size = os.path.getsize(file_path)
         file_metadata = {
             "path": file_path,
             "name": extract_file_name_from_path_string(file_path),
             "size": file_size,
         }
         with open(file_path, "rb") as f:
-            contents_metadata = DatasetMetadataReader(f.read()).read()
+            if not self.use_json:
```

Comment:
We should, implement an interface for dataset metadata readers, similar to how we have for data readers. This way we can call something like _metadata_reader_factory.get_service() and return the appropriate metadata reader.